language/move-lang/src/interface_generator.rs

```diff
@@ -129,6 +139,16 @@ impl<'a> Context<'a> {
 const DISCLAIMER: &str =
     "// NOTE: Functions are 'native' for simplicity. They may or may not be native in actuality.";
 
+fn write_friend_decl(ctx: &mut Context, fdecl: &ModuleHandle) -> String {
+    format!(
+        "friend {}::{}",
```

Comment:
I'd use the `uses` map so it is not always fully qualified... maybe?

The `uses` thing is designed to make types more readable. I could go either way for `friend`

Reply:
I am fine either way too but I tend to slightly favor having a qualified `friend`. The reason is because the `use 0x2::A;` will only show up in `friend A;`, and `A::<function-or-type>` will never show up in types or functions. Therefore, it is two lines `use 0x2::A;` + `friend A` vs one line `friend 0x2::A;`.

Reply:
I'll buy it. We can re-evaluate if needed as the feature gets used more